### PR TITLE
running the gradle script to produce the issue stack

### PR DESCRIPTION
**Pull request by tahiat**

### CODE REVIEW

**Comment by tahiat** [comment]
Work in progress PR

File: main.py [hunk 1]
```diff
@@ -336,27 +347,197 @@ def performEvaluation(issue_data) -> Result:
...
         result = run_specimin(issue_id ,specimin_command, os.path.join(issue_folder_abs_dir, specimin_project_name))
     
     print(f"{result.name} - {result.status}")
+
+    # build script is shipped with input program. It exists in the "specimin" directory of the input program's root directory.
+    # Coping the build script to the output directory of the minimized program.
+    build_script_path = os.path.join(issue_folder_abs_dir, issue_id, specimin_input, repo_name, specimin_project_name, "build.gradle")
+
+    if not os.path.exists(build_script_path):
+        return result
```
Comment (kelloggm) [inline main.py:378]: This should raise some kind of exception, should it not?

File: main.py [hunk 2]
```diff
@@ -336,27 +347,197 @@ def performEvaluation(issue_data) -> Result:
...
+    '''
+    Compare the error log of the minimized program with the expected error log
+    '''
+    with open(expected_log_path, "r") as file:
+        expected_content = file.read()
+
+    with open(actual_log_path, "r") as file:
+        actual_content = file.read()
+    
+   # bug_pattern_data = json.loads(bug_pattern_data)
```
Comment (kelloggm) [inline main.py:445]: Why is this commented out? Should it be deleted?

File: main.py [hunk 3]
```diff
@@ -336,27 +347,197 @@ def performEvaluation(issue_data) -> Result:
...
+
+    error_file = re.search(file_pattern, expected_content).group(1)
+    error_message = re.search(error_pattern, expected_content).group(1)
+    error_source = re.search(source_pattern, expected_content).group(1)
+    error_reason = re.search(reason_pattern, expected_content).group(1)
+
+    if error_file in actual_content and error_message in actual_content and error_source in actual_content and error_reason in actual_content:
+        return True
+    else:
+        return False
```
Comment (kelloggm) [inline main.py:460]: `return True else return False` is an anti-pattern. Instead, return the value of the guard.

File: main.py [hunk 4]
```diff
@@ -393,6 +574,10 @@ def main():
             evaluation_results.append(result)
             json_status[issue_id] = result.status
             print((f"{issue_id} <========= execution Ends."))
+            # task of crash generation on minimized program. 
+            # copy the build script from the input target specimin directory
+
```
Comment (kelloggm) [inline main.py:579]: Is something supposed to occur here? This comment seems to indicate that something should be done, but all that's been added here is some blank lines. If nothing is supposed to happen here, should we remove this comment?

File: resources/gradle/wrapper/gradle-wrapper.jar
Comment (kelloggm) [inline resources/gradle/wrapper/gradle-wrapper.jar:1]: Why do we need a copy of gradle in `specimin-evaluation`?

**Review by kelloggm — CHANGES_REQUESTED** [review]
Sorry for the delayed review.

File: resources/gradle/wrapper/gradle-wrapper.jar
Comment (tahiat) [reply resources/gradle/wrapper/gradle-wrapper.jar:1]: Minimized programs are built with gradle script for specific checker-framework version. Added the gradle wrapper to execute the gradle script.

Comment (kelloggm) [reply resources/gradle/wrapper/gradle-wrapper.jar:1]: Isn't the build script for each target project in that project's repository? In that case, why would we need the gradle wrapper here? The usual way to do this is to put the gradle wrapper in the target project.

Is the design here that every project shares the same gradle version?

File: main.py (hunk 1 above)
Comment (tahiat) [reply main.py:374]: Every target should have a build script. But I have not finished added build script for all. So if the build script is not present, I am just returning the result of minimization result of the target. 

After completion this should raise some flag to indicate that script is missing.

File: resources/gradle/wrapper/gradle-wrapper.jar
Comment (tahiat) [reply resources/gradle/wrapper/gradle-wrapper.jar:1]: Isn't the build script for each target project in that project's repository? -> Yes
 why would we need the gradle wrapper here? --> I copy the script in the minimized program's directory and execute the script to build minimized program (to produce the bug). The target project can be of different build system. 

Is the design here that every project shares the same gradle version? -> Yes, building all minimized program with the same gradle version.

Comment (tahiat) [reply resources/gradle/wrapper/gradle-wrapper.jar:1]: gradle build scripts are specifically to build the minimized program. a sample is provided below

```
plugins {
    id 'java-library'
    id 'org.checkerframework' version '0.6.37'
}

java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(11)
    }
}

apply plugin: 'org.checkerframework'

repositories {
    mavenCentral()
}

checkerFramework {
     checkers = [
            'org.checkerframework.checker.nullness.NullnessChecker'
    ]
    excludeTests = true
    incrementalize = false
}



dependencies {
  compileOnly 'org.checkerframework:checker-qual:3.0.1'
  checkerFramework 'org.checkerframework:checker:3.0.1'
}
```

Comment (kelloggm) [reply resources/gradle/wrapper/gradle-wrapper.jar:1]: These design decisions should be documented somewhere (not just in the PR description). I suggest that you write a short document that describes the steps to add a new target, which will force you to document this.

Comment (kelloggm) [reply resources/gradle/wrapper/gradle-wrapper.jar:1]: Let's merge this now and add a document describing the design in a separate PR.

**Pull request merged**